src/Symfony/Component/HttpFoundation/Request.php

```diff
@@ -382,7 +382,12 @@ public static function create(string $uri, string $method = 'GET', array $parame
      */
     public static function setFactory(?callable $callable): void
     {
-        self::$requestFactory = $callable;
+        if(null === $callable) {
+            self::$requestFactory = null;
+            return;
+        }
+
+        self::$requestFactory = $callable(...);
```

Comment:
We usually use a ternary expression when resolving this kind of union.

```suggestion
        if(null === $callable) {
            self::$requestFactory = null;
            return;
        }

        self::$requestFactory = null === $callable ? null : $callable(...);
```

Reply:
Fixed.